### DISCUSSION TITLE
Explainable AI - Ideas, Packages, Helpful Tips

### DISCUSSION THREAD

**Discussion by aruss175** [EAGE-Annual-Hackathon/EAGE-Annual-Hackathon-Discussion-Boards#2] [General]
This thread will provide with some ideas and help for getting started with this year's EAGE hackathon theme - Explainable AI.

How do we ensure trust that our models are behaving how they should?  How can we easily demonstrate this to those who don't have data science background?  Here are some examples to get your team thinking.

**Comment by aruss175** [comment]
A great place to start is with the two most common tools for ML explainability - LIME and SHAP:  here is a [nice overview](https://towardsdatascience.com/explainable-ai-xai-lime-shap-two-great-candidates-to-help-you-explain-your-machine-learning-a95536a46c4e).

**Reply by LukasMosser** [reply]
Here is another great video lecture by Michael Pyrcz on Shapely Values:
https://www.youtube.com/watch?v=oCIybnawLdg

There's a lot of detail in this one that might come as a handy resource!

**Comment by EdwinB12** [comment]
A potentially simple hackathon idea could be to essentially reproduce the Wolf/Husky [experiment](https://eage.eventsair.com/eageannual2022/hackathon-explainable-ai)  but with some geophysical data. 

For example, a network could be trained to simply classify seismic shot gathers are noisy or clean. Then using some explainable AI methods (LIME, Activation Maps, etc), some comparisons could be made between how geophysicists determine a noisy shot gather vs a model. Where does the ML model/Geophysicist agree/disagree, etc. 

This is my first hackathon so this may be laughably simple or stupidly ambitious or maybe just boring. . I'm not sure :)

**Comment by LukasMosser** [comment]
@EdwinB12 There's a repository that was recently made available for classifying unstructured data. This comes with a dataset and a pretrained model. Maybe this is something you where you could try the wolf husky experiment?

Here is the Github repository:
https://github.com/bolgebrygg/extract-and-classify-images-from-geoscience-documents

There are some great repositories out there to start from for heatmaps / Grad-CAM style visualizations. 
Ping me if you need some assistance with that.

**Comment by aruss175** [comment]
There is also the FORCE lithology prediction model that was done in 2020 - https://github.com/olawaleibrahim/2020_FORCE_Lithology_Prediction

**Reply by LukasMosser** [reply]
And here the corresponding dataset: 
https://drive.google.com/drive/folders/1GIkjq4fwgwbiqVQxYwoJnOJWVobZ91pL
